src/routes/posts.js

```diff
@@ -33,7 +33,37 @@ const createPost = async (req, res, next) => {
     }
 }
 
+// GET ALL ARTICLES
+
+const fetchPosts = async (req, res, next) => {
+    try {
+        const allArticles = await db.query('SELECT * FROM posts')
+
+        const articles = allArticles.rows
+        console.log(articles)
+        const { id, userId, title, image, content, published, createdAt } =
```

Comment:
yea i think this would be the cause. the articles is an array and not a plain old object  so we can't be destructure it like this

what you might want to do is to map over it

something like 

``` js
res.status(201).json({
             status: 'success',
             data: articles.map(article => {
                  id: articleId,
                  ...and so on
             })
                
         })
```

Reply:
kindly let me know if its clear and how it works out if it is

Reply:
it didn't work, i got a lot of errors. I have pushed it too

Reply:
![image](https://user-images.githubusercontent.com/68056104/185458817-9ef225b3-81d5-4367-b5e2-fb5ee007a51f.png)

Reply:
@beblicarl its different from what i sent though

Reply:
its not exact code i sent as well. kindly check out the [Array.map function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) and its usage

Reply:
![Screenshot (7)](https://user-images.githubusercontent.com/68056104/185462397-b0bd924e-dd23-4706-b940-bb176a25510e.png)
![Screenshot (8)](https://user-images.githubusercontent.com/68056104/185462402-6b6f7082-4ae5-4690-a3cd-a52811e9a07d.png)

When i mapped over the elements in the array and returned article it worked,
However, when i attempted storing the properties of article into their corresponding variables i got errors

![Screenshot (9)](https://user-images.githubusercontent.com/68056104/185462634-91852f84-d3d3-4216-8cb1-6cb23d364b24.png)

Reply:
@beblicarl got it. yea like i mentioned what i sent is just sort of an idea of how to go about it. its just a syntax error  that's happening like you see

Kindly push what you have. Also the screenshots don't convey much sometimes. mostly you should just have the code pushed so i can point out what is wrong directly on the line

Reply:
I am not allowed to push unless i comment places giving error messages out. Which i have just done

Reply:
okay, understood, no worries